shared-bindings/microcontroller/__init__.c

```diff
@@ -71,7 +71,11 @@
 //| .. method:: delay_us(delay)
 //|
 //|   Dedicated delay method used for very short delays. **Do not** do long delays
-//|   because it will stall any concurrent code.
+//|   because this stops all other functions from completing. Think of this as an empty
+//|   `while` loop that runs for the specified `(delay)` time. If you have other
+//|   code or peripherals that require specific timing or processing while you are waiting,
+//|   explore a different avenue such as using `time.monotonic()` to evaluate time
+//|   in a loop.
```

Comment:
Audio recording is one example that will not run during delay_us. You can also point to `time.sleep` as an alternative for longer delays because it will actually use something like `time.monotonic` to do some things in the background while waiting.

Reply:
Alright. I wasn't sure if `time.sleep` caused the same behavior or not.